File: Minimum-Absolute-Distance-Between-Mirror-Pairs.py
1class Solution:
2    def minMirrorPairDistance(self, nums: List[int]) -> int:
3        mindist = 100001
4        seen = {}
5        for index in range(len(nums)):
6            if(nums[index] in seen):
7                mindist = min(mindist,abs(seen.get(nums[index]) - index))
8            revnum = int(str(nums[index])[::-1])
9            seen[revnum] = index
10        return mindist if mindist!=100001 else -1

File: Minimum-Absolute-Distance-Between-Mirror-Pairs.java
1class Solution {
2private:
3    int revnum(int num){
4        int result = 0;
5        while(num){
6            result = result*10 + num%10;
7            num/=10;
8        }
9        return result;
10    }
11public:
12    int minMirrorPairDistance(vector<int>& nums) {
13        unordered_map<int,int> umap;
14        int mindist = INT_MAX;
15        for(int index = 0 ; index < nums.size() ; index++){
16            if(umap.contains(nums[index])){
17                mindist = min(mindist,abs(umap[nums[index]] - index));
18            }
19            umap[revnum(nums[index])] = index;
20        }
21        return (mindist!=INT_MAX ? mindist : -1);
22    }
23};